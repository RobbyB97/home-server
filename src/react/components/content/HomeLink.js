import React from 'react'
import {NavLink} from 'react-router-dom'


export default class HomeLink extends React.Component {

    render() {
        return (
            <div className='home-link'>
                {this.props.external ?
                    <a className="home-link__link" href={this.props.to}>
                        {this.props.icon &&
                            <img className="home-link__icon" src={this.props.icon} />
                        }

                        <p className="home-link__link-text">
                            {this.props.linkText ?
                                this.props.linkText
                            :
                                <p>HomeLink External</p>
                            }
                        </p>
                    </a>
                :
                    <NavLink className="home-link__link" to={this.props.to}>
                        {this.props.icon &&
                            <img className="home-link__icon" src={this.props.icon} />
                        }

                        <p className='home-link__link-text'>
                            {this.props.linkText ?
                                this.props.linkText
                            :
                                <p>HomeLink Internal</p>
                            }
                        </p>
                    </NavLink>
                }
            </div>
        )
    }
}