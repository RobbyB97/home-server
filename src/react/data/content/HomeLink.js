export const homeLink_Skills = {
    to: "/skills",
    linkText: "Skills"
}


export const homeLink_Experience = {
    to: "/experience",
    linkText: "Experience"
}


export const homeLink_Projects = {
    to: "/projects",
    linkText: "Projects"
}


export const homeLink_Resume = {
    to: "/resume",
    linkText: "Resume"
}