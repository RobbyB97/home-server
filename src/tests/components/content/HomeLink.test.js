import React from 'react'
import {shallow} from 'enzyme'

import HomeLink from '../../../react/components/content/HomeLink'


test('Render HomeLink', () => {
    const wrapper = shallow(<HomeLink />)

    expect(wrapper).toMatchSnapshot()
})