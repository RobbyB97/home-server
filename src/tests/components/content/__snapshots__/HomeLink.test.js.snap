// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render external HomeLink 1`] = `
<div
  className="homeLink"
>
  <a
    className="homeLink__link"
  >
    <p
      className="homeLink__link-text"
    >
      <p>
        HomeLink External
      </p>
    </p>
  </a>
</div>
`;

exports[`Render external HomeLink with icon 1`] = `
<div
  className="homeLink"
>
  <a
    className="homeLink__link"
  >
    <img
      className="homeLink__icon"
      src="icon!"
    />
    <p
      className="homeLink__link-text"
    >
      <p>
        HomeLink External
      </p>
    </p>
  </a>
</div>
`;

exports[`Render external HomeLink with linkText 1`] = `
<div
  className="homeLink"
>
  <a
    className="homeLink__link"
  >
    <p
      className="homeLink__link-text"
    >
      LinkText!
    </p>
  </a>
</div>
`;

exports[`Render internal HomeLink with icon 1`] = `
<div
  className="homeLink"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="homeLink__link"
  >
    <img
      className="homeLink__icon"
      src="icon!"
    />
    <p
      className="homeLink__link-text"
    >
      <p>
        HomeLink Internal
      </p>
    </p>
  </NavLink>
</div>
`;

exports[`Render internal HomeLink with linkText 1`] = `
<div
  className="homeLink"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="homeLink__link"
  >
    <p
      className="homeLink__link-text"
    >
      LinkText!
    </p>
  </NavLink>
</div>
`;

exports[`Render internal HomeLink without props 1`] = `
<div
  className="homeLink"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    className="homeLink__link"
  >
    <p
      className="homeLink__link-text"
    >
      <p>
        HomeLink Internal
      </p>
    </p>
  </NavLink>
</div>
`;
